ush/check_land_input_orography.py

```diff
@@ -0,0 +1,201 @@
+#!/usr/bin/env python3
+"""
+check_land_input_orography.py
+
+This script compares the land mask between input surface data files and orography files for consistency across tiles.
+
+It provides functions to:
+- Count valid points in 'vtype' and 'stype' variables from input NetCDF files.
+- Count land points in 'land_frac' variable from orography NetCDF files.
+- Compare the counts between input and orography data for each tile, optionally raising errors on mismatches.
+
+Functions
+---------
+count_points_from_input(input_dir: str) -> Dict
+
+count_points_from_orog(orog_dir: str) -> Dict
+    Counts the number of land points in orography data files for each tile, where 'land_frac' > 0.
+
+compare_counts(input_counts: Dict, orog_counts: Dict, fatal: bool = False) -> None
+    Compares the counts from input and orography data for each tile, logging warnings or raising errors on mismatches.
+
+Usage
+-----
+Run as a script with required arguments:
+    --input_dir : Path to input NetCDF files (sfc_data.tile{tile_num}.nc)
+    --orog_dir  : Path to orography NetCDF files (oro_data.tile{tile_num}.nc)
+    --fatal     : Optional flag to fail on first error
+
+Example
+-------
+python check_land_input_orography.py --input_dir /path/to/input --orog_dir /path/to/orog --fatal
+"""
+
+from argparse import ArgumentParser, ArgumentDefaultsHelpFormatter
+from logging import getLogger
+from typing import Dict
+import os
+from netCDF4 import Dataset
+import numpy as np
+from wxflow import Logger, logit, AttrDict
+
+logger = getLogger(__name__)
+NTILES = 6  # Number of tiles expected in the input and orography data
+
+
+@logit(logger)
+def count_points_from_input(input_dir: str) -> Dict:
+    """
+    Counts the number of valid points in the 'vtype' and 'stype' variables from surface data NetCDF files
+    in the specified input directory.
+    For each tile (from 1 to NTILES), attempts to open the corresponding 'sfc_data.tile{tt}.nc' file and
+    counts the number of elements greater than zero in the 'vtype' and 'stype' arrays.
```

Comment:
@GeorgeGayno-NOAA This is OK, right? We wouldn't expect any vtype=17 (water) in the restarts would we?